### ISSUE TITLE
[Feature Request] - Ability to mass edit links relative path

### ISSUE THREAD

**Issue by selra-hc** [giuspen/cherrytree#975]
Hi,

I'd like to have the possibility to mass edit relative path in my current ctd file.

I maintain a parallel folder architecture to store my documents outside the main file using relative path. I'd like to change those path to update my folder architecture when needed.
For example, folder \X\A may be changed to \X\Archives\2019\A. It means that all links to documents (file and folder) have to be updated if I want them available when clicking.

I tried to edit it directly in the file, but the path is encoded, e.g. <rich_text link="file UGVyc29ubmVsXE5vdGljZSB0w6lsw6lwaG9uZSBDaGFudGFsXE5vdGljZSBUw6lsw6lwaG9uZSBBbmRyb2lkLmRvY3g=">Notice</rich_text>

Is it possible to add a search and replace functionality for link's path ? If not, how would you suggest to do so without manually updating each one ?

Sorry if it's not the good way to ask, thanks for your help and keep up the good work (I really do enjoy the current functionalities that you provide us with cherrytree) !

**Comment by txe** [comment]
Do you know any script language (e.g. python)? Paths are encoded by Base64

**Comment by selra-hc** [comment]
Hi

Thanks for the hint txe. 

I can handle short scripts but it will take me some time, trials and errors :) but knowing about the Base64 (which may be obvious, but wasn't for me :) ) gave me some hindsight on a way to do it.

I made an attempt with vim using macros (select path, isolate it from the rest of the line, decode from base64, search pattern and replace, recode, put back the result in one line, and on to the next occurrence), it kinda worked but I have to test more tomorrow to see if it works as intended.

**Comment by ForeverRainbow** [comment]
@selra-hc Python has a b64 module as part of its standard library, and BeautifulSoup can be used to parse XML. Good learning experience if you want to try it :)

**Comment by selra-hc** [comment]
Hi ! Back from my experiments vith vim and base64 encode/decode. As said before, it kinda work ;)

TLDR; I almost succeeded, but not perfect due to padding in base64. However it meets it purpose as I don't have anymore to manually change the path for each link , but just to validate the edit prompt once when I need to acces a file.

@Forever64 BeautifulSoup seems indeed interesting for my purpose, and I'll try it later on ! But for my purpose Vim + macros is enough to do the job (although I have to admint it has been a complicated one, and if I had more links to convert, BeautifulSoup would be the more elegant and less time consuming way to go).

Starting from a link :
T2xkXEpvYlxIb21lXHJycg==
It decodes to : Old\Job\Home\rrr
I replace with : Old\Job\Archive\2019\Home\rrr (represents the manipulation that I usually do)
And I encode it with python base64.py to : T2xkXEpvYlxBcmNoaXZlXDIwMTlcSG9tZVxycnINCg==

This encoding doesn't exactly work back in Cherrytree (From what I understand, may be due to padding in base64). I get the error (01_error.png) enclosed. To correct this error, I just have to edit the link by opening the editor in cherrytree and validate without touching anything else in the dialog box (02_validation.png). Opening back the xml, the link changes to : T2xkXEpvYlxBcmNoaXZlXDIwMTlcSG9tZVxycnI= (4 character less).

The limit that I reach that way is that I don't see any way to change padding on encoding through base64.py, having the following options :
usage: C:\Users\Me\AppData\Local\Programs\Python\Python38\lib\base64.py [-d|-e|-u|-t] [file|-]
        -d, -u: decode
        -e: encode (default)
        -t: encode and decode string 'Aladdin:open sesame'
I suppose this is a limit I have from calling python in a vim shell where I don't have all options enabled, but that's a question I'll have t ask to a dedicated python forum :)

Anyway, while I still think it could be an interesting feature to add (though my use case may be unconventional and there are functionnalities that should be prioritised over ths), I am happy with what I have now (limited manpulation to update links and automated macros to do it when I want).

Thanks again @txe and @Forever64, sorry for being verbose, @txe please let me know if you'd like better to close the issue (or close it yourself) now or if it's a request that may be added later on on the cherrytree roadmap, and therefore to be left open for now :)

**Comment by selra-hc** [comment]
![01_error](https://user-images.githubusercontent.com/67684262/86462323-ba982d00-bd2b-11ea-923c-09a913fd99e5.png)
![02_validation](https://user-images.githubusercontent.com/67684262/86462332-be2bb400-bd2b-11ea-9aad-0beb08f9b2ed.png)

**Comment by selra-hc** [comment]
![01_error](https://user-images.githubusercontent.com/67684262/86462666-3e521980-bd2c-11ea-9942-614564f38b70.png)
![02_validation](https://user-images.githubusercontent.com/67684262/86462678-4447fa80-bd2c-11ea-92b6-00c64b68dc1f.png)

**Comment by txe** [comment]
I'm a bit surprise to see such path 'Old\Job\Home\rrr'. Is it right?

**Comment by selra-hc** [comment]
Yes it is :) rrr is a text file without extension, although I could have named it rrr.txt. For test purpose it was enough though, and this is what I have with all my files (mostly .docx and .odt)

![rrr](https://user-images.githubusercontent.com/67684262/86463624-27acc200-bd2e-11ea-8ba9-aee433222036.png)

**Comment by txe** [comment]
I will leave it. 
It actually is good idea to have ability to search and replace in links. The other things as tables, codeboxes, tags also have the same issue.
